delocate/tools.py

```diff
@@ -836,33 +857,34 @@ def dir2zip(in_dir, zip_fname):
 
     Parameters
     ----------
-    in_dir : str
+    in_dir : str or Path-like
         Directory path containing files to go in the zip archive
-    zip_fname : str
+    zip_fname : str or Path-like
         Filename of zip archive to write
+    compression : int, optional, keyword-only
+        The zipfile compression type used.
+    compress_level : int, optional, keyword-only
+        The compression level used for this archive.
     """
-    z = zipfile.ZipFile(zip_fname, "w", compression=zipfile.ZIP_DEFLATED)
-    for root, dirs, files in os.walk(in_dir):
-        for file in files:
-            in_fname = pjoin(root, file)
-            in_stat = os.stat(in_fname)
-            # Preserve file permissions, but allow copy
-            info = zipfile.ZipInfo(in_fname)
-            info.filename = relpath(in_fname, in_dir)
-            if os.path.sep == "\\":
-                # Make the path unix friendly on windows.
-                # PyPI won't accept wheels with windows path separators
-                info.filename = relpath(in_fname, in_dir).replace("\\", "/")
-            # Set time from modification time
-            info.date_time = time.localtime(in_stat.st_mtime)
-            # See https://stackoverflow.com/questions/434641/how-do-i-set-permissions-attributes-on-a-file-in-a-zip-file-using-pythons-zip/48435482#48435482 # noqa: E501
-            # Also set regular file permissions
-            perms = stat.S_IMODE(in_stat.st_mode) | stat.S_IFREG
-            info.external_attr = perms << 16
-            with open_readable(in_fname, "rb") as fobj:
-                contents = fobj.read()
-            z.writestr(info, contents, zipfile.ZIP_DEFLATED)
-    z.close()
+    with zipfile.ZipFile(
+        zip_fname, "w", compression=compression, compresslevel=compress_level
+    ) as zip:
+        for root, dirs, files in os.walk(in_dir):
+            for dir in dirs:
+                dir_path = Path(root, dir)
+                out_dir_name = str(dir_path.relative_to(in_dir)) + "/"
+                zip_info = zipfile.ZipInfo.from_file(dir_path, out_dir_name)
```

Comment:
I assume this normalizes the path to Unix style?

Reply:
I just now checked: Yes. `zipfile.ZipInfo.from_file` always calls [os.path.normpath](https://docs.python.org/3/library/os.path.html#os.path.normpath) on the path given.
> On Windows, it converts forward slashes to backward slashes.

Reply:
I accidentally read this backwards.  This would actually imply that it always converts to Windows separators on Windows.

Reply:
Once the archive path is passed to `ZipInfo.__init__` it converts the slashes:
```py
# This is used to ensure paths in generated ZIP files always use
# forward slashes as the directory separator, as required by the
# ZIP format specification.
if os.sep != "/" and os.sep in filename:
    filename = filename.replace(os.sep, "/")
```